File: bindings/lua/Polycode/AudioMixer.lua
class "AudioMixer"


function AudioMixer:__getvar(name)
	if name == "globalVolume" then
		return Polycode.AudioMixer_get_globalVolume(self.__ptr)
	elseif name == "listenerPosition" then
		local retVal = Polycode.AudioMixer_get_listenerPosition(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "listenerOrientation" then
		local retVal = Polycode.AudioMixer_get_listenerOrientation(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Quaternion"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
end

function AudioMixer:__setvar(name,value)
	if name == "globalVolume" then
		Polycode.AudioMixer_set_globalVolume(self.__ptr, value)
		return true
	elseif name == "listenerPosition" then
		Polycode.AudioMixer_set_listenerPosition(self.__ptr, value.__ptr)
		return true
	elseif name == "listenerOrientation" then
		Polycode.AudioMixer_set_listenerOrientation(self.__ptr, value.__ptr)
		return true
	end
	return false
end
function AudioMixer:AudioMixer(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.AudioMixer(unpack(arg))
	end
end

function AudioMixer:__delete()
	if self then Polycode.delete_AudioMixer(self.__ptr) end
end

File: bindings/javascript/Polycode/AudioMixer.js
function AudioMixer() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.AudioMixer()
	}
	Object.defineProperties(this, {
		'globalVolume': { enumerable: true, configurable: true, get: AudioMixer.prototype.__get_globalVolume, set: AudioMixer.prototype.__set_globalVolume},
		'listenerPosition': { enumerable: true, configurable: true, get: AudioMixer.prototype.__get_listenerPosition, set: AudioMixer.prototype.__set_listenerPosition},
		'listenerOrientation': { enumerable: true, configurable: true, get: AudioMixer.prototype.__get_listenerOrientation, set: AudioMixer.prototype.__set_listenerOrientation}
	})
}


AudioMixer.prototype.__get_globalVolume = function() {
	return Polycode.AudioMixer__get_globalVolume(this.__ptr)
}

AudioMixer.prototype.__set_globalVolume = function(val) {
	Polycode.AudioMixer__set_globalVolume(this.__ptr, val)
}

AudioMixer.prototype.__get_listenerPosition = function() {
	var retVal = new Vector3("__skip_ptr__")
	retVal.__ptr = 	Polycode.AudioMixer__get_listenerPosition(this.__ptr)
	return retVal
}

AudioMixer.prototype.__set_listenerPosition = function(val) {
	Polycode.AudioMixer__set_listenerPosition(this.__ptr, val.__ptr)
}

AudioMixer.prototype.__get_listenerOrientation = function() {
	var retVal = new Quaternion("__skip_ptr__")
	retVal.__ptr = 	Polycode.AudioMixer__get_listenerOrientation(this.__ptr)
	return retVal
}

AudioMixer.prototype.__set_listenerOrientation = function(val) {
	Polycode.AudioMixer__set_listenerOrientation(this.__ptr, val.__ptr)
}

Duktape.fin(AudioMixer.prototype, function (x) {
	if (x === AudioMixer.prototype) {
		return;
	}
	Polycode.AudioMixer__delete(x.__ptr)
})
